pkgs/applications/office/qownnotes/default.nix

```
@@ -52,23 +51,30 @@ stdenv.mkDerivation {
     "-DBUILD_WITH_SYSTEM_BOTAN=ON"
   ];
 
-  postInstall = ''
-    installShellCompletion --cmd ${appname} \
-      --bash <(xvfb-run $out/bin/${appname} --completion bash) \
-      --fish <(xvfb-run $out/bin/${appname} --completion fish)
-    installShellCompletion --cmd ${pname} \
-      --bash <(xvfb-run $out/bin/${appname} --completion bash) \
-      --fish <(xvfb-run $out/bin/${appname} --completion fish)
+  # Install shell completion on Linux (with xvfb-run)
+  postInstall = lib.optionalString stdenv.hostPlatform.isLinux ''
+      installShellCompletion --cmd ${appname} \
+        --bash <(xvfb-run $out/bin/${appname} --completion bash) \
+        --fish <(xvfb-run $out/bin/${appname} --completion fish)
+      installShellCompletion --cmd ${pname} \
+        --bash <(xvfb-run $out/bin/${appname} --completion bash) \
+        --fish <(xvfb-run $out/bin/${appname} --completion fish)
+  ''
+  # Install shell completion on macOS
+  + lib.optionalString stdenv.isDarwin ''
+      installShellCompletion --cmd ${pname} \
+        --bash <($out/bin/${appname} --completion bash) \
+        --fish <($out/bin/${appname} --completion fish)
   ''
   # Create a lowercase symlink for Linux
   + lib.optionalString stdenv.hostPlatform.isLinux ''
     ln -s $out/bin/${appname} $out/bin/${pname}
   ''
-  # Wrap application for macOS as lowercase binary
+  # Remame application for macOS as lowercase binary
```

```suggestion
  # Rename application for macOS as lowercase binary
```

(Also why is this needed?)

---

Thank you for your typo fix!

> (Also why is this needed?)

I'm not sure. In the original package, there was a step to create a lowercase binary. I was just trying to mimic that again. ️🤷🏻

Also, I don't know why no `.app` file falls out of the build process... Does anyone have experience with Darwin builds on NixOS?